infra/modules/common_environment/main.tf

```diff
@@ -0,0 +1,123 @@
+terraform {
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = ">= 3.111.0"
+    }
+  }
+}
+
+module "naming_convention" {
+  source = "../azure_naming_convention"
+
+  environment = {
+    prefix          = var.environment.prefix
+    env_short       = var.environment.env_short
+    location        = var.environment.location
+    domain          = var.environment.domain
+    app_name        = var.environment.app_name
+    instance_number = var.environment.instance_number
+  }
+}
+
+#------------------------#
+# COMMON RESOURCE GROUPS #
+#------------------------#
+resource "azurerm_resource_group" "common" {
+  name     = "${local.project}-common-rg-01"
+  location = var.environment.location
+
+  tags = var.tags
+}
+
+resource "azurerm_resource_group" "network" {
+  name     = "${local.project}-network-rg-01"
+  location = var.environment.location
+
+  tags = var.tags
+}
+
+resource "azurerm_resource_group" "test" {
```

Comment:
Shall who is going to do tests use this repo or any repo they like?

Reply:
The module creates the basic infrastructure, and this resource group is not structurally significant—it’s just a “suggestion”: similar to the _common_ resource group for shared resources, the _test_ one is provided for placing test resources, whether created through the portal or code.

It can be useful for those conducting tests without needing to create a dedicated resource group or for cleaning up old test resources in one go. However, since it’s just a logical grouping, we can remove it if it causes more confusion than value. The main risk is that without it, users testing resources might create them in the “common” or “network” resource groups for convenience.